orbiter_translations/automic/xml_demo.py

```diff
@@ -0,0 +1,359 @@
+    ...        ]
+    ...    }
+    ... ], 'SCRI': [
+    ...    {'@name': 'SCRI.SCRIPT', 'SCRIPT': [
+    ...            {'@mode': '1', '@state': '1', '#text': 'echo ""'
+    ...            }
+    ...        ]
+    ...    }
+    ... ]
+    ... })
+    ... # doctest: +ELLIPSIS
+    [{'@Alias': '', '@BranchType': '0', '@Col': '2', '@Lnr': '2', '@OH_TITLE': '', '@OType': 'SCRI', '@Object': 'SCRI.SCRIPT', '@Row': '1', '@State': '', '@Text1': '', 'predecessors': [{'pre': [{'@BranchType': '0', '@Lnr': '1', '@PreLnr': '1', '@When': '', '@type': 'container'}]}], 'script': {'@name': 'SCRI.SCRIPT', 'SCRIPT': [{'@mode': '1', '@state': '1', '#text': 'echo ""'}]}}]
+
+    ```
+    """  # noqa: E501
+    task_definitions = {}
+    tasks = []
+    for dag_struct_key, struct in val.items():
+        if dag_struct_key in ["SCRI", "JOBS"]:
+            for task_details in struct:
+                task_definitions[task_details["@name"]] = task_details
+
+    for dag_struct_key, struct in val.items():
+        if "JOBP" in dag_struct_key:
+            for task in struct[0]["JobpStruct"][0]["task"]:
+                if task["@OType"] in ["SCRI", "JOBS"]:
+                    task["script"] = task_definitions[task["@Object"]]
+                tasks.append(task)
+    return tasks
+
+
+@task_rule(priority=2)
+def basic_task_rule(
+    val: dict,
+) -> OrbiterEmptyOperator | OrbiterBashOperator | OrbiterSQLExecuteQueryOperator | None:
+    """Translate input into an Operator (e.g. `OrbiterBashOperator`)""
+
+    ```pycon
+    >>> basic_task_rule(val={'@Alias': '', '@BranchType': '0', '@Col': '2', '@Lnr': '2', '@OH_TITLE': 'generate_file', '@OType': 'SCRI', '@Object': 'SCRI.SCRIPT', '@Row': '1', '@State': '', '@Text1': '', 'predecessors': [
+    ...    {'pre': [
+    ...            {'@BranchType': '0', '@Lnr': '1', '@PreLnr': '1', '@When': '', '@type': 'container'
+    ...            }
+    ...        ]
+    ...    }
+    ... ], 'script': {'@name': 'SCRI.SCRIPT', 'SCRIPT': [
+    ...        {'@mode': '1', '@state': '1', '#text': 'echo "I am a Script"'
+    ...        }
+    ...    ]
+    ... }
+    ... })
+    ... # doctest: +ELLIPSIS
+    generate_file_task = BashOperator(task_id='generate_file', bash_command='echo "I am a Script"')
+
+    ```
+    """  # noqa: E501
+    try:
+        if val["@OType"] == "SCRI":
+            return OrbiterBashOperator(
+                task_id=val["@OH_TITLE"],
+                bash_command=val["script"]["SCRIPT"][0]["#text"],
+            )
+        if val["@OType"] == "JOBS":
+            if "SQL" in val["script"]["SCRIPT"]:
+                return OrbiterSQLExecuteQueryOperator(
+                    task_id=val["@OH_TITLE"],
+                    sql=val["script"]["SCRIPT"].split(":SQL")[-1].strip(),
+                    **conn_id(conn_id="mssql_default", conn_type="mssql"),
+                )
+        return OrbiterEmptyOperator(task_id=val["@OH_TITLE"])
+
+    except StopIteration:
+        pass
+    return None
```

Comment:
can you split these into three separate rules? One returning an `OrbiterEmptyOperator` (or modify the `cannot_map_rule`), one returning a `OrbiterSQLExecuteQueryOperator`, and one returning a `OrbiterBashOperator` ?

Reply:
Have split the task rules, as suggested.